cypress/integration/diagrams.spec.ts

```diff
@@ -12,42 +12,42 @@ describe('Diagram codeblock gets rendered: ', () => {
   it('markmap', () => {
     cy.codemirrorFill('```markmap\n- pro\n- contra\n```')
     cy.getMarkdownBody()
-      .find('div[data-cy=markmap] svg')
+      .find('[data-cy=markmap] svg')
       .should('be.visible')
   })
 
   it('vega-lite', () => {
     cy.codemirrorFill('```vega-lite\n{"$schema":"https://vega.github.io/schema/vega-lite/v4.json","data":{"values":[{"a":"","b":28}]},"mark":"bar","encoding":{"x":{"field":"a"},"y":{"field":"b"}}}\n```')
     cy.getMarkdownBody()
-      .find('div.vega-embed canvas')
+      .find('.vega-embed canvas')
       .should('be.visible')
   })
 
   it('graphviz', () => {
     cy.codemirrorFill('```graphviz\ngraph {\na -- b\n}\n```')
     cy.getMarkdownBody()
-      .find('div[data-cy=graphviz] svg')
+      .find('[data-cy=graphviz] svg')
       .should('be.visible')
   })
 
   it('mermaid', () => {
     cy.codemirrorFill('```mermaid\ngraph TD;\n    A-->B;\n```')
     cy.getMarkdownBody()
-      .find('div.mermaid > svg')
+      .find('.mermaid > svg')
       .should('be.visible')
   })
 
   it('flowchart', () => {
     cy.codemirrorFill('```flow\nst=>start: Start\ne=>end: End\nst->e\n```')
     cy.getMarkdownBody()
-      .find('div[data-cy=flowchart] svg')
+      .find('[data-cy=flowchart] svg')
       .should('be.visible')
   })
 
   it('abc', () => {
     cy.codemirrorFill('```abc\nM:4/4\nK:G\n|:GABc dedB:|\n```')
     cy.getMarkdownBody()
-      .find('div.abcjs-score > svg')
+      .find('.abcjs-score > svg')
       .should('be.visible')
   })
```

Comment:
I'm still not sure if it's a good idea to expect elements, that we don't control or create. The SVGs and the canvas you're looking for here are not part of our code, but of the 3rd-party rendering libs themself.
I understand, that you want to test if our code triggers the lib code but imho this is useless as long as we can't verify what the user is seeing. Also if e.g. abcjs switches from svg to canvas, then this test will fail and we still don't know if the output is correct.

Reply:
It's true that we don't control the specific elements. But we sometimes control the parent container that has the class name or data-cy attribute and show for example error messages inside it on failures.
When just testing for the parent container, we don't make sure that the library really runs. That's not end-to-end testing but end-to-one-step-before-end.
While it's also true that maybe one day a dependency changes their rendering method, it's actually good that the test fails in that case. It reminds us to check whether the rendering works still after the dependency restructuring. And changing one selector in a cypress test inside a renovate PR isn't a major deal.

Reply:
Tbh actually also doing an "end-to-one-step-before-end" test, because you're not checking if the lib renders the output correctly. To be a complete end to end test, you have to check if e.g. some words are contained in the output.

Reply:
We don't know if the svg you're checking contains any reasonable content. And as long as we don't check this, there is no point in checking the existence of the svg.

Reply:
Most of the libs probably have their own tests that their rendering output works. Therefore we don't need to test the libs that the use. We need to test whether we trigger the libs properly to render. When looking for a div that _could_ contain a diagram, it doesn't mean that the diagram library has been triggered. It also could be that the replacer was refactored but the lib call itself was forgotten.
When checking for the existence of just some rendering, we at least ensure that the rendering lib has been called.

Reply:
Theeeen how about "div > *" ? :thinking:

Reply:
In cypress it should be
```
      .children()
      .should("exist")
```

Reply:
I did this with the `.children()` method you mentioned.